src/analysis/processing/qgsalgorithmexecutespatialitequery.cpp

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************
+                         qgsalgorithmexecutepostgisquery.cpp
+                         ---------------------
+    begin                : May 2020
+    copyright            : (C) 2020 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmexecutespatialitequery.h"
+#include "qgsproviderregistry.h"
+#include "qgsprovidermetadata.h"
+#include "qgsabstractdatabaseproviderconnection.h"
+
+///@cond PRIVATE
+
+QString QgsExecuteSpatialiteQueryAlgorithm::name() const
+{
+  return QStringLiteral( "spatialiteexecutesql" );
+}
+
+QString QgsExecuteSpatialiteQueryAlgorithm::displayName() const
+{
+  return QObject::tr( "SpatiaLite execute SQL" );
+}
+
+QStringList QgsExecuteSpatialiteQueryAlgorithm::tags() const
+{
+  return QObject::tr( "database,sql,spatialite,execute" ).split( ',' );
+}
+
+QString QgsExecuteSpatialiteQueryAlgorithm::group() const
+{
+  return QObject::tr( "Database" );
+}
+
+QString QgsExecuteSpatialiteQueryAlgorithm::groupId() const
+{
+  return QStringLiteral( "database" );
+}
+
+QString QgsExecuteSpatialiteQueryAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Executes a SQL command on a SpatiaLite database." );
+}
+
+QgsExecuteSpatialiteQueryAlgorithm *QgsExecuteSpatialiteQueryAlgorithm::createInstance() const
+{
+  return new QgsExecuteSpatialiteQueryAlgorithm();
+}
+
+void QgsExecuteSpatialiteQueryAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterProviderConnection( QStringLiteral( "DATABASE" ), QObject::tr( "Database (connection name)" ), QStringLiteral( "spatialite" ) ) );
```

Comment:
This is a welcome move, but it's also an api break from the old algorithm, because the old approach allows executing commands on a non-connected db by file path.

Not sure what the best approach forward is here. @nirvn I believe we spoke about this recently too?

Reply:
My intention was to make it consistent with PostGIS algorithm which works only for registered connections. Also not sure which approach better.

Reply:
Me either. I think we possibly need two variants like some other algs have ("registered connections")...

Reply:
Ok, I made another algorithm for registered connections and removed API break.